pkg/controller.v2/controller_ref_manager.go

```diff
@@ -0,0 +1,398 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Note(CPH): this file is copied form k8s.io/kubernetes/pkg/controller
```

Comment:
how hard would it be to vendor this?

@ScorpioCPH

Reply:
We have no vendor `k8s.io/kubernetes/`, and CPH thinks that there is no need to import such a vendor for this struct.

Reply:
Yes, `k8s.io/kubernetes` is too large for us.